### PR TITLE
Workload Identity Support for AzureComponentFactory

### LINKED ISSUE #37943: [FEATURE REQ] Workload Identity Support for AzureComponentFactory
### Library name

Microsoft.Extensions.Azure

### Please describe the feature.

With the [deprecation of AAD Pod Identity](https://cloudblogs.microsoft.com/opensource/2022/01/18/announcing-azure-active-directory-azure-ad-workload-identity-for-kubernetes/), app developers working in Kubernetes have been migrating to [Azure Workload Identity](https://github.com/Azure/azure-workload-identity). The `Azure.Identity` library now supports a new `WorkloadIdentityCredential` as of version `1.9.0`, and it would great to take advantage of it using the `Microsoft.Extensions.Azure` library.

## Proposal
Add a new possible value for `credential` when creating credentials via `ClientFactory.CreateCredential` in addition to `"managedidentity"` called `"workloadidentity"`. This `credential` value reads settings from both the `IConfiguration` (or section) object as well as the [pre-defined environment variables](https://azure.github.io/azure-workload-identity/docs/quick-start.html#7-deploy-workload) to return a `WorkloadIdentityCredential` object.

## Workaround
Users can continue to use the existing `ManagedIdentityCredential` but instead use the environment variables to leverage the [token exchange](https://github.com/Azure/azure-sdk-for-net/blob/79fd982c0ef9944f113754ec9f17e393b2110f0e/sdk/identity/Azure.Identity/src/TokenExchangeManagedIdentitySource.cs#LL26C11-L26C11).

### PR DESCRIPTION
**Pull request by wsugarman**
Resolves #37943

- Allow `AzureCredentialFactory` to create instances of `WorkloadIdentityCredential` based on the new `"workloadidentity"` `credential` value
  - Workload Identity credentials must have a Tenant Id, Client Id, and a Token File Path specified
  - Add new `tokenFilePath` field to the configuration expected by `ClientFactory.CreateCredential`
- Update `Azure.Identity` version to `1.9.0` across the repository to support `WorkloadIdentityCredential`

### CODE REVIEW

**Review by jsquire — CHANGES_REQUESTED** [review]
Hi @wsugarman.  Thank you for your contribution and interest in improving the Azure developer experience.  As mentioned in #37943, I'd like to seek some perspectives from our Identity team before we make a decision on whether to move forward with these enhancements.

I'm going to mark as requesting changes in the meantime, to prevent any accidental approvals/merges.

**Comment by jsquire** [comment]
Hi @wsugarman.   Thanks for your patience.  I was able to find the right folks for perspective, and we're in agreement that this serves a core scenario and is an enhancement that we'd like to include.  If you're willing, would you please add an entry in the [changelog](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/CHANGELOG.md) that describes the new value?   Beyond that, it looks like we're good to merge.

**Comment by wsugarman** [comment]
@jsquire Awesome! I've added to the changelog (as well as added a comment in the code). Please let me know if you'd like to see any other changes.

File: sdk/extensions/Microsoft.Extensions.Azure/CHANGELOG.md [hunk 1]
```diff
@@ -4,12 +4,16 @@
...
 
+- Added support for creating `WorkloadIdentityCredential` objects from the configuration using the `"credential": "workloadidentity"`. Users must provide values for the `tenentId`, `clientId`, and newly added `tokenFilePath` keys in the configuration, or they must set the environment variables `AZURE_TENANT_ID`, `AZURE_CLIENT_ID`, and `AZURE_FEDERATED_TOKEN_FILE`.
+
 ### Breaking Changes
 
 ### Bugs Fixed
 
 ### Other Changes
 
+- Updated dependency `Azure.Identity` to version `1.9.0`
```
Comment (jsquire) [inline sdk/extensions/Microsoft.Extensions.Azure/CHANGELOG.md:15]:
```suggestion
- Updated dependency `Azure.Identity` to version `1.9.0`.
```

**Review by jsquire — APPROVED** [review]

**Comment by jsquire** [comment]
Thanks, @wsugarman.   Looks good.  I'll get this merged in and we'll look to release the new version as part of our August cycle next week.

**Pull request merged**